src/App.tsx

```diff
@@ -15,36 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = 'default',
+}
+
+interface FilterParams {
+  sortType: SortType;
+  reversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortType, reversed }: FilterParams,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.DEFAULT);
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortType,
+    reversed,
+  });
+  const reset = () => {
+    setSortType(SortType.DEFAULT);
+    setReversed(false);
+  };
```

Comment:
The function's name 'reset' should start with a verb that describes what the function does. Consider renaming it to better convey its purpose, for example 'resetFilters'. [CODE STYLE]